test/mocks/osx/MockPluginRepoRegistry.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity 0.8.17;
+
+import {IDAO} from "@aragon/osx/core/dao/IDAO.sol";
+import {InterfaceBasedRegistry} from "@aragon/osx/test/utils/InterfaceBasedRegistryMock.sol";
+import {IPluginRepo} from "@aragon/osx/framework/plugin/repo/IPluginRepo.sol";
+
+/// @title MockPluginRepoRegistry
+/// @author Aragon Association - 2022-2023
+/// @notice This contract maintains an address-based registry of plugin repositories in the Aragon App DAO framework.
+contract MockPluginRepoRegistry is InterfaceBasedRegistry {
+    /// @notice The ID of the permission required to call the `register` function.
+    // bytes32 public constant REGISTER_PLUGIN_REPO_PERMISSION_ID = keccak256("REGISTER_PLUGIN_REPO_PERMISSION");
+
+    /// @notice Emitted if a new plugin repository is registered.
+    /// @param subdomain The subdomain of the plugin repository.
+    /// @param pluginRepo The address of the plugin repository.
+    event PluginRepoRegistered(string subdomain, address pluginRepo);
+
+    // /// @notice Thrown if the plugin subdomain doesn't match the regex `[0-9a-z\-]`
+    // error InvalidPluginSubdomain(string subdomain);
+
+    // /// @notice Thrown if the plugin repository subdomain is empty.
+    // error EmptyPluginRepoSubdomain();
+
+    /// @dev Used to disallow initializing the implementation contract by an attacker for extra safety.
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @notice Initializes the contract by setting calling the `InterfaceBasedRegistry` base class initialize method.
+    /// @param _dao The address of the managing DAO.
+    function initialize(IDAO _dao) external initializer {
+        bytes4 pluginRepoInterfaceId = type(IPluginRepo).interfaceId;
+        __InterfaceBasedRegistry_init(_dao, pluginRepoInterfaceId);
+
+        // subdomainRegistrar = _subdomainRegistrar;
+    }
+
+    /// @notice Registers a plugin repository with a subdomain and address.
+    /// @param subdomain The subdomain of the PluginRepo.
+    /// @param pluginRepo The address of the PluginRepo contract.
+    function registerPluginRepo(string calldata subdomain, address pluginRepo) external 
+    // auth(REGISTER_PLUGIN_REPO_PERMISSION_ID)
+    {
+        // if (!(bytes(subdomain).length > 0)) {
+        //     revert EmptyPluginRepoSubdomain();
+        // }
+
+        // if (!isSubdomainValid(subdomain)) {
+        //     revert InvalidPluginSubdomain({subdomain: subdomain});
+        // }
+
+        // bytes32 labelhash = keccak256(bytes(subdomain));
+        // subdomainRegistrar.registerSubnode(labelhash, pluginRepo);
+
+        // _register(pluginRepo);
+
+        // emit PluginRepoRegistered(subdomain, pluginRepo);
+    }
```

Comment:
for our internal tooling (this PR is fine, just making a note) I think we could uncomment most of this other than the _register and .registerSubnode lines which at the very least will throw if subdomains are invalid (if we care about testing this)

Reply:
The ultimate solution would be for ENS's to be optional. Other than that, we may test modified contracts of OSx, but I wouldn't spend too much energy here

Reply:
Yeah agreed on all fronts